Add contributor set-up instructions (#471)
wikimedia/ws-export: README.md

  * `fonts-gubbi`
  * `fonts-opendyslexic`

## Development installation (without Docker)

> This section deals with a full installation of WS Export on the local development machine. For Docker installation instructions, see below.

Reviewer: ```suggestion
This section deals with a full installation of WS Export on the local development machine. For Docker installation instructions, see [below](#docker-developer-environment).
```

1. Get the source code:

   ```console
   git clone https://github.com/wikimedia/ws-export.git
   cd ws-export
   ```

2. Install dependencies:

   ```console
   composer install
   ```

   A few PHP extensions are required; if these are not installed, Composer will let you know
   and you will need to install them with your operating system's package manager.
   For example, in Debian-based Linux distributions:

   ```console
   apt install php-sqlite3 php-zip php-curl
   ```

   Then create a `.env.local` file:

   ```console
   cp .env .env.local
   ```

   Edit `.env.local` and set `APP_ENV=dev` and `APP_SECRET` to a random string of your choice.

3. [Install Symfony CLI](https://symfony.com/download#step-1-install-symfony-cli)
   and start the development web server with:

   ```console
   symfony server:start -d
   ```

4. Open your web browser to e.g. <http://localhost:8000>

>    and the basic operations should be working with the following requirements:

Reviewer: I think "limitations" may be a better word choice than "requirements". WDYT?

Author: Done!

   * you can only export to EPUB format; and
   * you must check the 'Exclude credits' option,
     to avoid querying the database for usernames.

   Continue on for setting up more exporting and development options.

5. Install optional dependencies:

   **Calibre:**
   In order to export to PDF, plain text, RTF, or Mobi formats
   you must install [Calibre](https://calibre-ebook.com)
   so that the tool can use the `ebook-convert` command.

   **epubcheck:**

>    To validate exported ebooks (with the `./bin/console app:check` command),

Reviewer: ```suggestion
   To validate exported ebooks (with the [`./bin/console app:check` command](#appcheck)),
```

By the way, if the `./bin/console` is needed to run the command, why isn't it included in the "CLI Usage" section below? :thinking:

   you must install [epubcheck](https://github.com/w3c/epubcheck).

>    If it's not installed at `/usr/bin/epubcheck` then
>    set the `EPUBCHECK_JAR` environment variable.

Reviewer: ```suggestion
   If it installed at a location other than `/usr/bin/epubcheck`, then
   set the `EPUBCHECK_JAR` environment variable to the correct path.
```

   **Fonts:**
   To run the integration tests, also install
   the `fonts-linuxlibertine` package.
   You can also install any other fonts you want to use for exporting books.

6. WS Export uses two database connections:
   firstly, to its own database to store download statistics;
   and secondly, it connects to the Wikimedia database replicas
   to fetch information about Wikisource contributors
   for the credits list that can be included at the end of exported books.

>    For the statistics recording database you create the the database

Reviewer: ```suggestion
   For the statistics recording database you need to create the the database
```

   and add connection details to `.env.local`:

   ```console
   DATABASE_URL=mysql://db_user:db_password@127.0.0.1:3306/db_name
   ```

7. Then create the database tables with:

   ```
   ./bin/console doctrine:migrations:migrate`
   ```

8. To connect to the Wikimedia replicas, this tool uses
   the [Toolforge Bundle](https://github.com/wikimedia/ToolforgeBundle),
   and connects to [multiple databases](https://github.com/wikimedia/ToolforgeBundle#replicas-connection-manager).

>    Set replicas credentials in your `.env.local` file according to the Bundle's documentation.

Reviewer: ```suggestion
   Set the replicas' credentials in your `.env.local` file
   according to the Bundle's documentation.
```

9. Establish an SSH tunnel to the replicas:

  ```console
  ./bin/console toolforge:ssh
  ```

10. At this point, you should be able to use all of the functionality

>     of WS Export both via the web interface and CLI.

Reviewer: ```suggestion
    of WS Export, via both the web interface and the CLI.
```

   for the credits list that can be included at the end of exported books.

   For the statistics recording database you create the the database
   and add connection details to `.env.local`:

   ```console
   DATABASE_URL=mysql://db_user:db_password@127.0.0.1:3306/db_name
   ```

7. Then create the database tables with:

   ```
   ./bin/console doctrine:migrations:migrate`
   ```

8. To connect to the Wikimedia replicas, this tool uses
   the [Toolforge Bundle](https://github.com/wikimedia/ToolforgeBundle),
   and connects to [multiple databases](https://github.com/wikimedia/ToolforgeBundle#replicas-connection-manager).

   Set replicas credentials in your `.env.local` file according to the Bundle's documentation.

9. Establish an SSH tunnel to the replicas:

  ```console
  ./bin/console toolforge:ssh
  ```

10. At this point, you should be able to use all of the functionality
    of WS Export both via the web interface and CLI.

## CLI Usage

### app:check

Run epubcheck on books. With no options set, this will check 10 random books from English Wikisource.
Note that the random 10 will be cached (for repeatability) unless you use <info>--nocache</info>.

> ```console

Author: @waldyrious I didn't notice this before, sorry. This section of CLI usage is auto-generated (by [this thing](https://packagist.org/packages/samwilson/console-readme-generator)), so shouldn't be changed by hand. I thought we had a CI check in place to guard against this; I'm adding such a thing in #476.

Reviewer: Sure, no worries. Would you like me to revert the changes to the "CLI Usage" section, and keep the changes to the other sections?

Reviewer: In the meantime, I submitted https://github.com/samwilson/console-readme-generator/pull/7 to change this formatting in the source.

Author: Thanks! And no, it's fine to leave them as-is here, I'll merge that change and then we can update the package here.
